test-programs/sdk-test/src/lib.rs

```diff
@@ -0,0 +1,296 @@
+    name: String,
+) -> ProgramResult {
+    let light_accounts = inputs
+        .accounts
+        .as_ref()
+        .ok_or(LightSdkError::ExpectedAccounts)?;
+
+    let address_merkle_context = light_accounts[0]
+        .address_merkle_context
+        .ok_or(LightSdkError::ExpectedAddressMerkleContext)?;
+    let address_merkle_context = unpack_address_merkle_context(address_merkle_context, accounts);
+    let (address, address_seed) = derive_address(
+        &[b"compressed", name.as_bytes()],
+        &address_merkle_context,
+        program_id,
+    );
+
+    let my_compressed_account = LightAccountInfo::from_meta_init(
+        &light_accounts[0],
+        MyCompressedAccount::discriminator(),
+        address,
+        address_seed,
+        Some(name.len() + mem::size_of::<NestedData>()),
+        program_id,
+    )?;
+
+    let cpda = MyCompressedAccount {
+        name,
+        nested: NestedData::default(),
+    };
+    let data = my_compressed_account
+        .data
+        .ok_or(LightSdkError::ExpectedData)?;
+    let mut data = data.borrow_mut();
+    let mut write_cursor = Cursor::new(&mut *data);
+    cpda.serialize(&mut write_cursor)?;
+
+    let accounts = &mut accounts.iter();
+    let fee_payer = next_account_info(accounts)?;
+    let authority = next_account_info(accounts)?;
+    let registered_program_pda = next_account_info(accounts)?;
+    let noop_program = next_account_info(accounts)?;
+    let account_compression_authority = next_account_info(accounts)?;
+    let account_compression_program = next_account_info(accounts)?;
+    let invoking_program = next_account_info(accounts)?;
+    let light_system_program = next_account_info(accounts)?;
+    let system_program = next_account_info(accounts)?;
+
+    let light_cpi_accounts = LightCpiAccounts {
+        fee_payer,
+        authority,
+        registered_program_pda,
+        noop_program,
+        account_compression_authority,
+        account_compression_program,
+        invoking_program,
+        light_system_program,
+        system_program,
+        cpi_context: None,
+    };
+    verify_light_account_infos(
+        light_cpi_accounts,
+        &[],
+        inputs.proof,
+        &[my_compressed_account],
+        None,
+        false,
+        None,
+    )?;
+
+    Ok(())
+}
+
+fn update_nested_data(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    inputs: LightInstructionData,
+    nested_data: NestedData,
+) -> ProgramResult {
+    let light_accounts = inputs
+        .accounts
+        .as_ref()
+        .ok_or(LightSdkError::ExpectedAccounts)?;
+
+    let my_compressed_account = LightAccountInfo::from_meta_mut(
+        &light_accounts[0],
+        MyCompressedAccount::discriminator(),
+        program_id,
+    )?;
+
+    let data = my_compressed_account
+        .data
+        .ok_or(LightSdkError::ExpectedData)?;
+    let mut data = data.borrow_mut();
+    let mut cpda = MyCompressedAccount::deserialize(&mut data.as_slice())?;
```

Comment:
Since we are using borsh here, can we do a test which uses `LightAccount`directly (just duplicate this one but use `LightAccount` without anchor?
(It definitely makes sense to have a test without `LightAccount` as this one to make sure that the sdk works with any deserialization.)

Reply:
what do you think about renaming `LightAccount` to `LightBorshAccount` ?
(This would fit nicely with future `LightZeroCopyAccount`.)

Reply:
I like the idea.

The reason why I went for `LightAccount` before is sticking with Anchor's naming. But I can see the appeal of implementing more wrappers (bytemuck, bincode etc.).